controllers/front/SmailyCustomerCron.php

```diff
@@ -81,21 +84,21 @@ private function syncContacts()
                     $userdata = $this->getUserData($customer);
                     array_push($update_data, $userdata);
                 }
-                // Send subscribers to Smaily.
-                $response = $this->module->callApi('contact', $update_data, 'POST');
-                // Response logging.
-                if (isset($response['success'])) {
-                    $response = $response['result']['message'];
-                } else {
-                    $response =  $response['error'];
+
+                $chunks = array_chunk($update_data, self::SUBSCRIBERS_BATCH_LIMIT);
+                foreach ($chunks as $chunk) {
+                    // Send subscribers to Smaily.
+                    $response = $this->module->callApi('contact', $chunk, 'POST');
+                    // Response logging in case of error.
+                    if (isset($response['result']['code']) && $response['result']['code'] !== 101) {
+                        $this->module->logTofile('smaily-cron.txt', 'Customer sync failed - ' . $response['message']);
+                        break;
```

Comment:
Would you like for the synchronization run to pass? I mean `break` only stops the loop from running, but CRON run will still exist with status 0 (OK) and `User synchronization done!` is logged.

It would be better that in case of an error CRON run exists with non-zero status code. Then developers can use this as an indication that synchronization failed.

Reply:
Yes, I was also thinking that it requests all customers to a single array may cause problems with memory allocation. Now I limit query and iterate over it with offset and limit values. https://github.com/sendsmaily/smaily-prestashop-module/pull/10/commits/fa51248d09a0cfbd37f339b0208444869b1f62ac